src/event-stream/event-stream.ts

```diff
@@ -6,48 +6,36 @@ import {
   CoreNodeNakamotoBlockMessage,
   StackerDbChunk,
 } from './core-node-message';
-import { logger as defaultLogger, stopwatch } from '@hirosystems/api-toolkit';
+import { logger as defaultLogger, stopwatch, WorkerThreadManager } from '@hirosystems/api-toolkit';
 import { ENV } from '../env';
-import {
-  ParsedNakamotoBlock,
-  ParsedStackerDbChunk,
-  parseNakamotoBlockMsg,
-  parseStackerDbChunk,
-} from './msg-parsing';
+import { ParsedNakamotoBlock, ParsedStackerDbChunk } from './msg-parsing';
 import { SignerMessagesEventPayload } from '../pg/types';
-import { ThreadedParser } from './threaded-parser';
 import { SERVER_VERSION } from '@hirosystems/api-toolkit';
 import { EventEmitter } from 'node:events';
-
-// TODO: move this into the @hirosystems/salt-n-pepper-client lib
-function sanitizeRedisClientName(value: string): string {
-  const nameSanitizer = /[^!-~]+/g;
-  return value.trim().replace(nameSanitizer, '-');
-}
+import * as msgParserWorkerBlocks from './msg-parser-worker-blocks';
+import * as msgParserWorkerStackerDb from './msg-parser-worker-stackerdb';
 
 export class EventStreamHandler {
   db: PgStore;
   logger = defaultLogger.child({ name: 'EventStreamHandler' });
   eventStream: StacksEventStream;
-  threadedParser: ThreadedParser;
+  threadedParserBlocks = new WorkerThreadManager(msgParserWorkerBlocks);
+  threadedParserStackerDb = new WorkerThreadManager(msgParserWorkerStackerDb);
```

Comment:
TODO: multiple WorkerThreadManagers creates a lot of memory overhead from more worker threads. Should use a single worker that can parse either msg type.